### PR TITLE
Changelog for v2.167.0

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,79 @@
...
+-------------------------
+- Update the ``StormPkgTest.getTestCore()`` API to add a ``preppkghook``
+  callback option. This can be used to execute code prior to loading Storm
+  packages. The ``getTestCore()`` API now waits for ``onload`` handlers to
+  complete for each package it loads.
+  (`#3687 <https://github.com/vertexproject/synapse/pull/3687>`_)
+- Ensure that the ``Cell.ahaclient`` is fully owned and managed by the
+  ``Cell``. It will no longer use a global client that may exist.
+  (`#3677 <https://github.com/vertexproject/synapse/pull/3677>`_)
+- Update the ``stix2-validator`` library constraints to ``3.2.0,<4.0.0``.
```
Comment (MichaelSquires) [inline CHANGELOG.rst:47]:
```suggestion
- Update the ``stix2-validator`` library constraints to ``>=3.2.0,<4.0.0``.
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,6 +6,79 @@
...
+- TBD
+
+Deprecations
+------------
+- Deprecate the use of ``hiveboot.yaml`` to configure a Cell hive. This will be
+  removed 2024-05-05.
+  (`#3678 <https://github.com/vertexproject/synapse/pull/3678>`_)
+- The Telepath APIs for interacting with a Cell Hive, ``listHiveKey``,
+  ``getHiveKeys``, ``getHiveKey``, ``setHiveKey``, ``popHiveKey``, and
+  ``saveHiveTree`` have been deprecated. The tools ``synapse.too.s.hive.load``
```
Comment (MichaelSquires) [inline CHANGELOG.rst:77]:
```suggestion
  ``saveHiveTree`` have been deprecated. The tools ``synapse.tools.hive.load``
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -6,6 +6,79 @@
...
+=====================
+
+Model Changes
+-------------
+- Updates to the ``base`` model.
+  (`#3674 <https://github.com/vertexproject/synapse/pull/3674>`_)
+
+  **Deprecated Types**
+
+  The following forms have been marked as deprecated:
```
Comment (Cisphyx) [inline CHANGELOG.rst:19]:
```suggestion
  The following types have been marked as deprecated:
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -6,6 +6,82 @@
...
+
+Bugfixes
+--------
+- Asyncio Tasks created by signal handlers on the Base object are now held
+  onto, to ensure that they cannot be garbage collected before or during
+  their task execution.
+  (`#3686 <https://github.com/vertexproject/synapse/pull/3686>`_)
+- Update the ``Axon.postfiles`` and ``Axon.wput`` APIs to check for the
+  existence of files before making an attempting to send them over an
+  HTTP connection.
```
Comment (Cisphyx) [inline CHANGELOG.rst:60]:
```suggestion
  existence of files before attempting to send them over an HTTP connection.
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -6,6 +6,82 @@
...
+  (`#3682 <https://github.com/vertexproject/synapse/pull/3682>`_)
+- Fix an issue where pruning a non-existent tag mistakenly pruned related
+  tags.
+  (`#3673 <https://github.com/vertexproject/synapse/pull/3673>`_)
+
+Improved Documentation
+----------------------
+- TBD
```
Comment (Cisphyx) [inline CHANGELOG.rst:68]: Are we expecting any documentation updates in this release?

File: CHANGELOG.rst [hunk 6]
```diff
@@ -6,6 +6,82 @@
...
+
+Deprecations
+------------
+- Deprecate the use of ``hiveboot.yaml`` to configure a Cell hive. This will be
+  removed 2024-05-05.
```
Comment (Cisphyx) [inline CHANGELOG.rst:73]:
```suggestion
  removed on 2024-05-05.
```

File: CHANGELOG.rst [hunk 7]
```diff
@@ -6,6 +6,82 @@
...
+  (`#3678 <https://github.com/vertexproject/synapse/pull/3678>`_)
+- The Telepath APIs for interacting with a Cell Hive, ``listHiveKey``,
+  ``getHiveKeys``, ``getHiveKey``, ``setHiveKey``, ``popHiveKey``, and
+  ``saveHiveTree`` have been deprecated. The tools ``synapse.tool.s.hive.load``
```
Comment (Cisphyx) [inline CHANGELOG.rst:77]:
```suggestion
  ``saveHiveTree`` have been deprecated. The tools ``synapse.tools.hive.load``
```

File: CHANGELOG.rst [hunk 8]
```diff
@@ -6,6 +6,82 @@
...
+  * ``graph:event``
+  * ``edge:refs``
+  * ``edge:has``
+  * ``edge:wentto``
+  * ``graph:edge``
+  * ``graph:timeedge``
+
+Features and Enhancements
+-------------------------
+- Update the ``StormPkgTest.getTestCore()`` API to add a ``preppkghook``
```
[the remark this answers is not in the document]
Comment (vEpiphyte) [reply CHANGELOG.rst:39]: We cannot have `-` in argument names.

File: CHANGELOG.rst (hunk 5 above)
Comment (vEpiphyte) [reply CHANGELOG.rst:68]: From #3685

File: CHANGELOG.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (Cisphyx) [reply CHANGELOG.rst:39]: I know that, i was just writing it like that to clarify what i was asking

[the remark this answers is not in the document]
Comment (vEpiphyte) [reply CHANGELOG.rst:39]: We could change it to `prepkghook` and avoid the repetition?

File: CHANGELOG.rst [hunk 9]
```diff
@@ -6,6 +6,115 @@
...
+-------------
+- Updates to the ``base`` and ``file`` models.
+  (`#3674 <https://github.com/vertexproject/synapse/pull/3674>`_)
+  (`#3688 <https://github.com/vertexproject/synapse/pull/3688>`_)
+
+  **Updated Types**
+
+  ``file:path``
+    Normalizing paths such as ``../.././..`` previously failed. This now
+    produces a empty path.
```
Comment (Cisphyx) [inline CHANGELOG.rst:29]:
```suggestion
    produces an empty path.
```

File: CHANGELOG.rst [hunk 10]
```diff
@@ -6,6 +6,115 @@
...
+  * ``graph:edge``
+  * ``graph:timeedge``
+
+Features and Enhancements
+-------------------------
+- Add ``aha.svc.list`` and ``aha.svc.stat`` commands to enumerate the AHA
+  services. Add ``$lib.aha`` Storm APIs to delete, get, and list the AHA
+  services.
+  (`#3685 <https://github.com/vertexproject/synapse/pull/3685>`_)
+- Add a ``protected`` option to that can be set on Views to prevent
```
Comment (Cisphyx) [inline CHANGELOG.rst:57]:
```suggestion
- Add a ``protected`` option that can be set on Views to prevent
```

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**